src/transport_picoquic.cpp

```diff
@@ -531,34 +514,25 @@ PicoQuicTransport::setStatus(TransportStatus status)
 StreamId
 PicoQuicTransport::createStream(const TransportContextId& context_id, bool use_reliable_transport)
 {
-    logger.log(LogLevel::debug, "App requests to create new stream");
-
     const auto& iter = active_streams.find(context_id);
     if (iter == active_streams.end()) {
         throw std::invalid_argument("Invalid context id, cannot create stream");
     }
 
-    const auto& cnx_stream_iter = iter->second.find(0);
+    const auto datagram_stream_id = ::make_datagram_stream_id(_is_server_mode, _is_unidirectional);
+    const auto& cnx_stream_iter = iter->second.find(datagram_stream_id);
     if (cnx_stream_iter == iter->second.end()) {
-        throw std::invalid_argument("Missing primary connection zero stream, cannot create streams");
+        throw std::logic_error("Missing primary connection zero stream, cannot create streams");
     }
 
     if (!use_reliable_transport)
-        return ::make_datagram_stream_id(_is_server_mode, _is_bidirectional);
-
-    next_stream_id = ::make_stream_id(next_stream_id + 4, _is_server_mode, _is_bidirectional);
-
-    PicoQuicTransport::StreamContext *stream_cnx = createStreamContext(
-      cnx_stream_iter->second.cnx,
-      next_stream_id);
+        return datagram_stream_id;
 
-    TransportContextId tcid = context_id;
-    StreamId stream_id = next_stream_id;
+    next_stream_id += 4;
```

Comment:
Just incrementing by 4 since we set the server/direction bits on construction. Of course, using `::make_stream_id` would protect against accidental increments of those bits but we're unlikely to do it here, so..... it's fine.